include/dgl/immutable_graph.h

```diff
@@ -517,7 +500,7 @@ class COO : public GraphInterface {
   aten::COOMatrix adj_;
 
   /*! \brief whether the graph is a multi-graph */
-  LazyObject<bool> is_multigraph_;
+  Lazy<bool> is_multigraph_;
```

Comment:
Why we need to make it lazy here?

Reply:
Yes. Checking whether the graph is a multigraph is costly, and since the graph is immutable, we could postpone the check till the first time the flag is requested.

Reply:
I see. I didn't understand the implementation before. It seems like a cache more than a lazy initialized object to me.

Reply:
yes